test/stores/VDocument.test.ts

```diff
@@ -1088,5 +1088,223 @@ describe('stores', () => {
                 });
             });
         });
+        describe('deleteSelection', () => {
+            it('should delete part of the text within a paragraph', () => {
+                // deleteForward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p>ab[cd]ef</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>ab[]ef</p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p>ab]cd[ef</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>ab[]ef</p>',
+                });
+                // deleteBackward
+                // Forward
+                testEditor({
+                    contentBefore: '<p>ab[cd]ef</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>ab[]ef</p>',
+                });
+                // Backward
+                testEditor({
+                    contentBefore: '<p>ab]cd[ef</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>ab[]ef</p>',
+                });
+            });
+            it('should delete across two paragraphs', () => {
+                // deleteForward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p>ab[cd</p><p>ef]gh</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>ab[]gh</p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p>ab]cd</p><p>ef[gh</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>ab[]gh</p>',
+                });
+                // deleteBackward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p>ab[cd</p><p>ef]gh</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>ab[]gh</p>',
+                });
+                // Backward selection
+                testEditor({
+                    contentBefore: '<p>ab]cd</p><p>ef[gh</p>',
+                    stepFunction: deleteBackward,
+                    contentAfter: '<p>ab[]gh</p>',
+                });
+            });
+            it('should delete all the text in a paragraph', () => {
+                // deleteForward
+                // Forward selection
+                testEditor({
+                    contentBefore: '<p>[abc]</p>',
+                    stepFunction: deleteForward,
+                    contentAfter: '<p>[]<br></p>',
```

Comment:
Just wondering: Here we have the range right before the BR. Is it intended or is it a side-effect of our current implementation ? Would it make more or less sense to have `<p><br/>[]</p>` ?

Reply:
The range is always _before_ a visible BR (with the occasional exception of a trailing BR). This one is not visible so it doesn't technically matter - it's a side-effect of our current implementation that favors a range _before_ a BR. Semantically I'd argue there is no difference whatsoever given that that BR has no real meaning itself, but is a quirk of the DOM's implementation (nothing would stand in the way of saying a P has a height no matter what - you can even do that with CSS).